hack/tools/boskosctl/main.go

```diff
@@ -0,0 +1,384 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/netip"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"go4.org/netipx"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"sigs.k8s.io/cluster-api-provider-vsphere/hack/tools/pkg/boskos"
+	"sigs.k8s.io/cluster-api-provider-vsphere/hack/tools/pkg/janitor"
+)
+
+var (
+	boskosHost           string
+	resourceOwner        string
+	resourceType         string
+	resourceName         string
+	vSphereUsername      string
+	vSpherePassword      string
+	vSphereServer        string
+	vSphereTLSThumbprint string
+	vSphereFolder        string
+	vSphereResourcePool  string
+)
+
+func main() {
+	log := klog.Background()
+	ctx := ctrl.LoggerInto(context.Background(), log)
+	// Just setting this to avoid that CR is complaining about a missing logger.
+	ctrl.SetLogger(log)
+
+	rootCmd := setupCommands(ctx)
+
+	if err := rootCmd.Execute(); err != nil {
+		log.Error(err, "Failed running boskosctl")
+		os.Exit(1)
+	}
+}
+
+func setupCommands(ctx context.Context) *cobra.Command {
+	// Root command
+	rootCmd := &cobra.Command{
+		Use:          "boskosctl",
+		SilenceUsage: true,
+		Short:        "boskosctl can be used to consume Boskos vSphere resources",
+	}
+	// Note: http://boskos.test-pods.svc.cluster.local is the URL of the service usually used in k8s.io clusters.
+	rootCmd.PersistentFlags().StringVar(&boskosHost, "boskos-host", getOrDefault(os.Getenv("BOSKOS_HOST"), "http://boskos.test-pods.svc.cluster.local"), "Boskos server URL.")
+	rootCmd.PersistentFlags().StringVar(&resourceOwner, "resource-owner", "", "Owner for the resource.")
+
+	// acquire command
+	acquireCmd := &cobra.Command{
+		Use:  "acquire",
+		Args: cobra.NoArgs,
+		RunE: runCmd(ctx),
+	}
+	acquireCmd.PersistentFlags().StringVar(&resourceType, "resource-type", "", "Type of the resource. Should be one of: vsphere-project-cluster-api-provider, vsphere-project-cloud-provider, vsphere-project-image-builder")
+	rootCmd.AddCommand(acquireCmd)
+
+	// heartbeat command
+	heartbeatCmd := &cobra.Command{
+		Use:  "heartbeat",
+		Args: cobra.NoArgs,
+		RunE: runCmd(ctx),
+	}
+	heartbeatCmd.PersistentFlags().StringVar(&resourceName, "resource-name", "", "Name of the resource.")
+	rootCmd.AddCommand(heartbeatCmd)
+
+	// release command
+	releaseCmd := &cobra.Command{
+		Use:  "release",
+		Args: cobra.NoArgs,
+		RunE: runCmd(ctx),
+	}
+	releaseCmd.PersistentFlags().StringVar(&resourceName, "resource-name", "", "Name of the resource.")
+	releaseCmd.PersistentFlags().StringVar(&vSphereUsername, "vsphere-username", os.Getenv("VSPHERE_USERNAME"), "vSphere username of the resource, required for cleanup before release")
+	releaseCmd.PersistentFlags().StringVar(&vSpherePassword, "vsphere-password", os.Getenv("VSPHERE_PASSWORD"), "vSphere password of the resource, required for cleanup before release")
```

Comment:
```suggestion
	releaseCmd.PersistentFlags().StringVar(&vSpherePassword, "vsphere-password", "", "vSphere password of the resource, required for cleanup before release")
```
and read the env var down below otherwise if by any chance we run help we can leak the password

As alternative, let's simply drop sensitive flags (password, thumbprint) and rely on env vars

Reply:
I'll read the env vars below. I would prefer keeping the flag. Env vars is just such a strange API in my opinion :D

Reply:
+1 for keeping the flags.